### PR TITLE
press and hold implementation

### PR DESCRIPTION
**Pull request by thiccaxe**

### CODE REVIEW

**Comment by thiccaxe** [comment]
Still need to clean up some console logs, a bit of typescript work and look into docs.

File: src/primitives/useFocusManager.ts [hunk 1]
```diff
@@ -99,10 +96,51 @@ const keyMapEntries: Record<string | number, string> = {
   Escape: 'Escape',
 };
 
+const keyHoldMapEntries: Record<string | number, string> = {
```
Comment (chiefcll) [inline src/primitives/useFocusManager.ts:99]: This is cool - so users would specify which keys to check for hold - good optimization...

File: src/primitives/useFocusManager.ts [hunk 2]
```diff
@@ -154,45 +207,86 @@ export const useFocusManager = (userKeyMap?: Partial<KeyMap>) => {
...
+            elm[`on${mappedEvent}` as keyof KeyMapEventHandlers];
+          if (isFunc(onKeyHandler)) {
+            if (onKeyHandler.call(elm, e, elm, finalFocusElm) === true) {
               break;
             }
           }
+        } else {
+          console.log(`Unhandled key event: ${e.key || e.keyCode}`);
+        }
+        const fallbackFunction = isHold ? elm.onKeyPress : elm.onKeyHold;
```
Comment (chiefcll) [inline src/primitives/useFocusManager.ts:231]: Are these reversed? isHold => onKeyHold?

**Review by chiefcll — COMMENTED** [review]
This LGTM

**Comment by CosmaVergari** [comment]
Can the signal returned by `createKeyHold` from solid-primitives be of any help for this?

**Comment by thiccaxe** [comment]
Perhaps, but I don't think it would simplify the logic?
From what I can tell, we would have to diff the previous held keys and the current held keys, etc.

**Comment by CosmaVergari** [comment]
I think that a single signal for each key can be created -> [Code Reference](https://github.com/solidjs-community/solid-primitives/blob/5e1ade9e9ba7ab6f186e0c3b43e070739b1963ec/packages/keyboard/src/index.ts#L251)
So maybe we can internally map the keyHoldMap to signals for each key?

**Comment by thiccaxe** [comment]
Yes, that would likely be possible.

However, it should be noted that the current API expects a keyboard event. This would mean a small refactor. I don't know if this is actually used anywhere though. This would mean a small refactor.

**Pull request merged**